package/base-files/files/lib/upgrade/common.sh

```diff
@@ -110,14 +110,19 @@ get_magic_vfat() {
 	(get_image "$@" | dd bs=1 count=3 skip=54) 2>/dev/null
 }
 
+get_magic_fat32() {
+	(get_image "$@" | dd bs=1 count=5 skip=82) 2>/dev/null
+}
+
 part_magic_efi() {
 	local magic=$(get_magic_gpt "$@")
 	[ "$magic" = "EFI PART" ]
 }
 
 part_magic_fat() {
 	local magic=$(get_magic_vfat "$@")
-	[ "$magic" = "FAT" ]
+	local magic_fat32=$(get_magic_fat32 "$@")
+	[ "$magic" = "FAT" ] || [ "$magic_fat32" = "FAT32" ]
```

Comment:
I would recommend using `-o` instead of `||` here...

Reply:
Fixed

Reply:
Um, actually `-o` is deprecated in POSIX `sh`.